### PR TITLE
fix: improve key rotation

### PR DESCRIPTION
**Pull request by livio-a**
requires https://github.com/caos/oidc/pull/66

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -83,3 +83,5 @@ require (
 	k8s.io/apiextensions-apiserver v0.18.5
 	k8s.io/apimachinery v0.18.5
 )
+
+replace github.com/caos/oidc => /Users/livio/workspaces/go/src/github.com/caos/oidc
```
Comment (fgerschwiler) [inline go.mod:87]: remove and update oidc version

File: internal/auth/repository/eventsourcing/eventstore/key.go [hunk 2]
```diff
@@ -2,27 +2,58 @@ package eventstore
...
+	"github.com/caos/zitadel/internal/key/model"
+	key_view "github.com/caos/zitadel/internal/key/repository/view"
+	"github.com/caos/zitadel/internal/v2/command"
 )
 
 type KeyRepository struct {
-	View               *view.View
-	SigningKeyRotation time.Duration
+	Commands                 *command.CommandSide
+	Eventstore               *eventstore.Eventstore
```
Comment (fgerschwiler) [inline internal/auth/repository/eventsourcing/eventstore/key.go:24]: Do you use eventstore v2 on purpose?

File: internal/auth/repository/eventsourcing/eventstore/key.go [hunk 3]
```diff
@@ -40,17 +71,98 @@ func (k *KeyRepository) GetKeySet(ctx context.Context) (*jose.JSONWebKeySet, err
...
+			k.lockID, err = id.SonyFlakeGenerator.Next()
+			logging.Log("EVENT-bsdf6").OnError(err).Panic("unable to generate lockID")
+		}
+		context.TODO()
+	}
+	return k.lockID
+}
+
+func (k *KeyRepository) getKeyEvents(ctx context.Context, sequence uint64) ([]eventstore.EventReader, error) {
+	return k.Eventstore.FilterEvents(ctx, key_view.KeyPairQuery(sequence))
```
Comment (fgerschwiler) [inline internal/auth/repository/eventsourcing/eventstore/key.go:167]: did you use eventstore 2 on purpose?

File: internal/auth/repository/eventsourcing/eventstore/key.go (hunk 2 above)
Comment (livio-a) [reply internal/auth/repository/eventsourcing/eventstore/key.go:24]: which else should i use? v1 will be removed / deprecated when new-eventstore will be merged into master

File: internal/auth/repository/eventsourcing/eventstore/key.go (hunk 3 above)
Comment (livio-a) [reply internal/auth/repository/eventsourcing/eventstore/key.go:167]: which else should i use? v1 will be removed / deprecated when new-eventstore will be merged into master

**Review by fgerschwiler — APPROVED** [review]

**Pull request merged**